### ISSUE TITLE
search does not catch common abbreviations for portuguese roads

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#4955]
**[Submitted to the original trac issue database at 12.31pm, Saturday, 17th August 2013]**

When searching for portuguese streets, these abbreviations are common:

r = rua
d = dom
av = avenida

for example, Rua Dom Carlos de Mascarenhas in Lisboa has the tag 
"name = Rua D. Carlos de Mascarenhas". Searching for it under this name will find the correct place: http://www.openstreetmap.org/browse/way/233144763

The following search terms do not work, and should work:
Rua Dom Carlos de Mascarenhas
R Dom Carlos de Mascarenhas
R D Carlos de Mascarenhas

The abbraviation av does seem to work properly. The following two search terms both return the correct street http://www.openstreetmap.org/browse/way/8053005

av da liberdade, lisboa
avenida da liberdade, lisboa

I have conferred with a list of common abbrevs to find some more, which you might consider including as well, into the substitution list of the search engine:

Al. alameda
cal. calada 
Est. estrada 
L. largo
P. praa

Basically everything listed on this page, with "(toponimicamente)" beside it is a common place name abbreviation, but the above are the ones that I would recognise and use:

https://pt.wikipedia.org/wiki/Anexo:Lista_de_abreviaturas
